### ISSUE TITLE
Unable to run unit tests that reference MsalService

### ISSUE THREAD

**Issue by jlilly** [AzureAD/microsoft-authentication-library-for-js#6596]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

3.3.0

### Wrapper Library

MSAL Angular (@azure/msal-angular)

### Wrapper Library Version

3.0.6

### Public or Confidential Client?

Public

### Description

Unable to run unit tests due to a missing file in the release. Downgraded to 3,2,0 (and @azure/msal-angular@3.0.5) and the unit test successfully runs.

jest@28.1.3
jest-preset-angular@12.2.3

Angular CLI: 15.2.8
Node: 18.12.1
Package Manager: npm 8.19.2
OS: win32 x64

Angular: 15.2.9
... animations, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1502.8
@angular-devkit/build-angular   15.2.8
@angular-devkit/core            15.2.8
@angular-devkit/schematics      15.2.8
@angular/cli                    15.2.8
@schematics/angular             15.2.8
rxjs                            7.8.1
typescript                      4.8.4

### Error Message

  ● Test suite failed to run

    Cannot find module './index-9800158f.js' from 'node_modules/@azure/msal-browser/lib/msal-browser.cjs'

    Require stack:
      node_modules/@azure/msal-browser/lib/msal-browser.cjs
      node_modules/@azure/msal-angular/fesm2020/azure-msal-angular.mjs
      src/app/auth/auth.state.spec.ts

### Msal Logs

Not applicable

### MSAL Configuration

```javascript
Not Applicable
```

### Relevant Code Snippets

```javascript
See stack trace
```

### Reproduction Steps

`$ jest`

### Expected Behavior

Jest runs

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

None (Server), Other

### Regression

@azure/msal-browser 3.2.0

### Source

Internal (Microsoft)

**Comment by TomONeill** [comment]
Same problem here, running with Jest.

**Comment by shinjilow** [comment]
Same problem here, running with msal-react with Jest.

**Comment by miroberes** [comment]
Same here, Jest
 Require stack:
      node_modules/@azure/msal-browser/lib/msal-browser.cjs
      node_modules/msal-react-tester/dist/MsalReactTester.js
      node_modules/msal-react-tester/dist/index.js
      src/setupTests.js

**Comment by thillebrandt** [comment]
Yup, here too. 
`@azure/msal-angular: 3.0.6`
`@angular/*: 16.2.10`
`jest: 29.7.0`

**Comment by submarines-and** [comment]
Affects us when importing MSAL in watch mode

![image](https://github.com/AzureAD/microsoft-authentication-library-for-js/assets/3114694/29a2a45b-002b-4b29-81e2-115d09c7c49c)

**Comment by kai-dorschner-twinsity** [comment]
This issue happens to us, too. We're using babel for tests that uses the .cjs version internally. After debugging I found several problems in your compile target:

### node_modules/@azure/msal-browser/lib/msal-browser.cjs
```js
/*! @azure/msal-browser v3.3.0 2023-10-20 */
'use strict';
'use strict';

var index = require('./index-9800158f.js');



exports.AccountEntity = index.AccountEntity;
exports.ApiId = index.ApiId;
exports.AuthError = index.AuthError;
exports.AuthErrorCodes = index.AuthErrorCodes;
exports.AuthErrorMessage = index.AuthErrorMessage;
exports.AuthenticationHeaderParser = index.AuthenticationHeaderParser;
exports.AuthenticationScheme = index.AuthenticationScheme;
exports.AzureCloudInstance = index.AzureCloudInstance;
exports.BrowserAuthError = index.BrowserAuthError;
exports.BrowserAuthErrorCodes = index.BrowserAuthErrorCodes;
exports.BrowserAuthErrorMessage = index.BrowserAuthErrorMessage;
exports.BrowserCacheLocation = index.BrowserCacheLocation;
exports.BrowserConfigurationAuthError = index.BrowserConfigurationAuthError;
exports.BrowserConfigurationAuthErrorCodes = index.BrowserConfigurationAuthErrorCodes;
exports.BrowserConfigurationAuthErrorMessage = index.BrowserConfigurationAuthErrorMessage;
exports.BrowserPerformanceClient = index.BrowserPerformanceClient;
exports.BrowserPerformanceMeasurement = index.BrowserPerformanceMeasurement;
exports.BrowserStorage = index.BrowserStorage;
exports.BrowserUtils = index.BrowserUtils;
exports.CacheLookupPolicy = index.CacheLookupPolicy;
exports.ClientAuthError = index.ClientAuthError;
exports.ClientAuthErrorCodes = index.ClientAuthErrorCodes;
exports.ClientAuthErrorMessage = index.ClientAuthErrorMessage;
exports.ClientConfigurationError = index.ClientConfigurationError;
exports.ClientConfigurationErrorCodes = index.ClientConfigurationErrorCodes;
exports.ClientConfigurationErrorMessage = index.ClientConfigurationErrorMessage;
exports.DEFAULT_IFRAME_TIMEOUT_MS = index.DEFAULT_IFRAME_TIMEOUT_MS;
exports.EventMessageUtils = index.EventMessageUtils;
exports.EventType = index.EventType;
exports.InteractionRequiredAuthError = index.InteractionRequiredAuthError;
exports.InteractionRequiredAuthErrorCodes = index.InteractionRequiredAuthErrorCodes;
exports.InteractionRequiredAuthErrorMessage = index.InteractionRequiredAuthErrorMessage;
exports.InteractionStatus = index.InteractionStatus;
Object.defineProperty(exports, 'InteractionType', {
	enumerable: true,
	get: function () { return index.InteractionType; }
});
Object.defineProperty(exports, 'LogLevel', {
	enumerable: true,
	get: function () { return index.LogLevel; }
});
exports.Logger = index.Logger;
exports.MemoryStorage = index.MemoryStorage;
exports.NavigationClient = index.NavigationClient;
exports.OIDC_DEFAULT_SCOPES = index.OIDC_DEFAULT_SCOPES;
exports.PerformanceEvents = index.PerformanceEvents;
exports.PromptValue = index.PromptValue;
exports.ProtocolMode = index.ProtocolMode;
exports.PublicClientApplication = index.PublicClientApplication;
exports.PublicClientNext = index.PublicClientNext;
exports.ServerError = index.ServerError;
exports.ServerResponseType = index.ServerResponseType;
exports.SignedHttpRequest = index.SignedHttpRequest;
exports.StringUtils = index.StringUtils;
exports.UrlString = index.UrlString;
exports.WrapperSKU = index.WrapperSKU;
exports.stubbedPublicClientApplication = index.stubbedPublicClientApplication;
exports.version = index.version;
//# sourceMappingURL=msal-browser.cjs.map
```

you've got a duplicate `'use strict';` statement, and a reference to the missing file `var index = require('./index-9800158f.js');`

**Comment by kebryansg** [comment]
I managed to solve it temporarily by configuring the file ` jest.config.js`

`module.exports = {
  moduleNameMapper: {
    ...,
    '@azure/msal-browser': '<rootDir>/node_modules/@azure/msal-browser/lib/msal-browser.js',
  },
}`

**Comment by ImagoDeo** [comment]
I'm also running into this problem. Bumping from 3.1.0 -> 3.3.0 is not urgent, so I'll wait for a patch.

**Comment by gestj** [comment]
I also have this issue (when running my unit tests with Jest) ...

```
Cannot find module './index-9800158f.js' from '../../node_modules/@azure/msal-browser/lib/msal-browser.cjs' 
```

@kai-dorschner-twinsity posted already the deep dive into code I also did. I wonder how this could happen where that mentioned `index-9800158f.js` should be. 

@kebryansg Can you please post complete example of your `jest.config.js`?

I tried out your proposed workaround. But, it doesn't work for me. Maybe I did it wrong... Or it's different for me anyway. I have a mono repo setup with NX and a typescript version of jest config => `jest.config.ts`.

My **(not working)** version (should be fine from my point of view considering also official Jest documentation about `moduleNameMapper`):

```
const { getJestProjects } = require('@nx/jest');

export default {
    moduleNameMapper: {
        '@azure/msal-browser': '<rootDir>/node_modules/@azure/msal-browser/lib/msal-browser.js',
    },
    projects: getJestProjects()
};
```


https://jestjs.io/docs/configuration#modulenamemapper-objectstring-string--arraystring

**Comment by gestj** [comment]
I also downgraded to `@azure/msal-angular@3.0.4` & `@azure/msal-browser@3.1.0` for now.

Those versions don't have this issue.

**Comment by TomONeill** [comment]
Seems fixed in [@azure/msal-browser@3.4.0](https://www.npmjs.com/package/@azure/msal-browser/v/3.4.0) and [@azure/msal-angular@3.0.7](https://www.npmjs.com/package/@azure/msal-angular/v/3.0.7)

Thank you!
